uwblueprint/sistema
# add user mailing lists

# [Change Email Subscriptions](https://www.notion.so/uwblueprintexecs/Change-Email-Subscriptions-49a7ae1740944c0fa085d720d47d96c6)



## Implementation description

COMMIT 1 - schema change: added email subscriptions 
- migration to create many-to-many relationship
- update seeding script to reflect this change

COMMIT 2 - types
- prettify interfaces and names and stuff

COMMIT 3 - other stuff
- update frontend to add "email subscriptions" tab
- edit /users endpoint to intake `getMailingLists` 



## Steps to test

1. [apply migration](https://www.notion.so/uwblueprintexecs/Change-Email-Subscriptions-49a7ae1740944c0fa085d720d47d96c6)
2. build and start project (`docker compose build --up`)
3. ensure the seeding command worked correctly:

Add a console.log on the results of the /users endpoint. (in `profile.tsx`, add `console.log(users)` after line 18)
Should look like this, containing the names of the lists.
![image](https://github.com/user-attachments/assets/a98e373c-118c-48d3-8f66-f5773498f458)

Run:
`docker exec -it sistema-db-1 psql -U sistema -d sistema`
`SELECT * FROM "MailingList";`
`SELECT * FROM "User";`

Each mailing list should have a list of some number (1-5) of emails that match with the emails of the users. This should also match the results of the /users endpoint in the /manage page.
![image](https://github.com/user-attachments/assets/0a02aa1a-dc83-466a-88ea-38e8d1312bdf)

4. ensure /profile works as expected: should look like:

![image](https://github.com/user-attachments/assets/fadf7528-025d-4817-b17d-2858dc4c4cae)
5. ensure /manage works as expected. (see screenshot in 3. seeding script testing)




## What should reviewers focus on?

1. seeding script - does the logic make sense?
2. types interface - I personally believe in it because both `manage.tsx` and `profile.tsx` define interfaces for `User`. Where should it be defined otherwise? Hard to decide IMO, so a central `types` seems like an ok resolution.

## Checklist
